warehouse/malware/checks/example.py

```diff
@@ -0,0 +1,32 @@
+from warehouse.malware.checks.base import MalwareCheckBase
+from warehouse.malware.models import (
+    MalwareVerdict,
+    VerdictClassification,
+    VerdictConfidence,
+)
+
+VERSION = 1
+SHORT_DESCRIPTION = "An example hook-based check"
+LONG_DESCRIPTION = """The purpose of this check is to demonstrate the implementation \
+of a hook-based check. This check will generate verdicts if enabled."""
+
+
+class ExampleCheck(MalwareCheckBase):
+
+    version = VERSION
+    short_description = SHORT_DESCRIPTION
+    long_description = LONG_DESCRIPTION
+
+    def __init__(self, check_id):
+        super().__init__()
+        self.id = check_id
+
+    def run(self, request, file_id):
+        verdict = MalwareVerdict(
+            check_id=self.id,
+            file_id=file_id,
+            classification=VerdictClassification.benign,
+            confidence=VerdictConfidence.High,
+            message="Nothing to see here!",
+        )
+        request.db.add(verdict)
```

Comment:
As a general design question: What do you think about performing the database transactions behind the scenes and exposing an imperative API? Something like this:

```python3
class MalwareCheckBase:
  def __init__(self):
    self._verdicts = []

  def verdict(self, **kwargs):
    self._verdicts.append(MalwareVerdict(**kwargs))

  def perform(self, ...):
    self.run(...)
    request.db.add_all(self._verdicts)

  ...
```

(With `perform` being the new top-level method submitted to the task queue)

With individual checks doing something like:

```python3
def run(self, request, file_id):
  self.verdict(...)
```

(or `deliver_verdict` or similar if just `verdict` isn't imperative enough)

The main advantages to this (that I can think of) is that it hides the involvement of the DB from check implementors, and gives us the ability to do additional bookkeeping (or future refactoring) within the `verdict` method.

Reply:
Yes! I like that a lot. This PR is just a PoC, but this is definitely the direction that I want to go in.